### PR TITLE
Fix GenomeDiagram division by zero Error & some flake8 style

### PR DESCRIPTION
**Pull request by sssxie**
This pull request addresses issue #...

- [ ] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [ ] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Graphics/GenomeDiagram/_LinearDrawer.py [hunk 1]
```diff
@@ -982,22 +993,19 @@ def draw_line_graph(self, graph):
...
         start, end = self._current_track_start_end()
         data = graph[start:end]
 
         # midval is the value at which the x-axis is plotted, and is the
         # central ring in the track
-        if graph.center is None:
-            midval = (maxval + minval) / 2.
-        else:
-            midval = graph.center
+        midval = graph.center if graph.center else (maxval + minval) / 2.
```
Comment (peterjc) [inline Bio/Graphics/GenomeDiagram/_LinearDrawer.py]: This is a functional change, what if ``graph.center`` was zero?

File: Bio/Graphics/GenomeDiagram/_LinearDrawer.py [hunk 2]
```diff
@@ -1118,15 +1126,11 @@ def draw_bar_graph(self, graph):
...
-            datarange = trackheight
+        datarange = datarange if datarange else trackheight
         data = graph[self.start:self.end]
         # midval is the value at which the x-axis is plotted, and is the
         # central ring in the track
-        if graph.center is None:
-            midval = (maxval + minval) / 2.
-        else:
-            midval = graph.center
+        midval = graph.center if graph.center else (maxval + minval) / 2.
```
Comment (peterjc) [inline Bio/Graphics/GenomeDiagram/_LinearDrawer.py]: This is a functional change, what if ``graph.center`` was zero?

File: Bio/Graphics/GenomeDiagram/_LinearDrawer.py [hunk 3]
```diff
@@ -1141,20 +1145,17 @@ def draw_bar_graph(self, graph):
...
             fragment1, x1 = self.canvas_location(pos1)
             x0, x1 = self.x0 + x0, self.x0 + x1     # account for margin
             barval = trackheight * (val - midval) / resolution
-            if barval >= 0:  # Different colors for bars that extend above...
-                barcolor = graph.poscolor
-            else:           # ...or below the axis
-                barcolor = graph.negcolor
-
+            # Different colors for bars that extend above...
+            # ...or below the axis
```
Comment (peterjc) [inline Bio/Graphics/GenomeDiagram/_LinearDrawer.py]: With a single line of code now, there is no reason to split the comment anymore, e.g.

```python
# Different colors for bars that extend above/below the axis
```

**Comment by peterjc** [comment]
I made some minor comments on the style changes, but I didn't see the division by zero fix - was there a second commit missing, or did I just read this too quickly?

Also, is it easy to add a unit case for the divide by zero condition?

**Comment by peterjc** [comment]
OK - could you include a unit test? If you have a tiny example which triggered the original divide by zero error, that could be good starting point.

**Comment by sssxie** [comment]
> OK - could you include a unit test? If you have a tiny example which triggered the original divide by zero error, that could be good starting point.

Since I didn't write a log before, this could take a long time. I found this problem in batch drawing. I'm updating the annotation of a species, using this module to show the new and old structural changes of each gene, about 300k genes. Although I used multiprocessing, I have to rerun the old version before this change for a long time to find out which gene caused the error.

**Comment by peterjc** [comment]
Are you able to take a look at this @widdowquinn ?
